app/html/dmx.html

```diff
@@ -2,11 +2,158 @@
+    <div class="section">
+        <input id="boxNumber" type="text" value="0"/>
+        <button id='boxOn'>Box On</button>
+        <button id='boxOff'>Box Off</button>
+    </div>
+    <br>
+
+    <!-- All channels -->
+    <div class="section">
+        <button id='allOn'>All On</button>
+        <button id='allOff'>All Off</button>
+        <button id='fadeOn'>Fade On</button>
+        <button id='fadeOff'>Fade Off</button>
+    </div>
+    <br>
+
+
 </body>
+    <script type="text/javascript" src="../js/dmx.js"></script>
+    <script>
+        /* Fading setup */
+        var isFading = false;
+        var currentFade = 0;
+        const fadeDuration = 5000;
+        const fadeInterval = 50;
+        var fadeChange = 255.0 * fadeInterval / fadeDuration;
+
+        setInterval(function() {
+            if (isFading) {
+                currentFade += fadeChange;
+
+                if (currentFade < 0) {
+                    currentFade = 0;
+                    fadeChange = -fadeChange;
+                }
+                if (currentFade > 255) {
+                    currentFade = 255;
+                    fadeChange = -fadeChange;
+                }
+
+                // Divide by 1 to make it an int
+                setAll(parseInt(currentFade));
+            }
+        }, fadeInterval);
+
+        // Returns the value for the element with the given id, parsed as an int.
+        // If failed to parse or find, returns default value
+        function getIntFromElementById(id, defaultVal) {
+            const val = parseInt(document.getElementById(id).value);
+
+            if (isNaN(val)) {
+                alert("Hey, " + val + " isn't an int! (" + id + ")");
+                return defaultVal;
+            }
+
+            return val;
+        }
+
+        /* Single channels */
+
+        document.getElementById('channelOn').onclick = function () {
+            const channel = getIntFromElementById("singleChannel", -1);
+            if (channel != -1) {
+                setChannel(channel, 255);
+            }
+        };
+        document.getElementById('channelOff').onclick = function () {
+            const channel = getIntFromElementById("singleChannel", -1);
+            if (channel != -1) {
+                setChannel(channel, 0);
+            }
+        };
+
+        /* Range of channels */
+
+        document.getElementById('rangeOn').onclick = function () {
+            const start = getIntFromElementById("rangeStart", -1);
+            if (start == -1) return;
+
+            const end = getIntFromElementById("rangeEnd", -1);
+            if (end == -1) return;
+
+            setRange(start, end, 255);
+        };
+        document.getElementById('rangeOff').onclick = function () {
+            const start = getIntFromElementById("rangeStart", -1);
+            if (start == -1) return;
+
+            const end = getIntFromElementById("rangeEnd", -1);
+            if (end == -1) return;
+
+            setRange(start, end, 0);
```

Comment:
Should this prompt the user if start > end or either of them are negative numbers?

Reply:
Probably. Will add that check